pkg/inference/backends/llamacpp/llamacpp_config.go

```diff
@@ -16,9 +17,34 @@ type Config struct {
 	Args []string
 }
 
+// isTerminal returns true if the given file is a character device (a terminal).
+func isTerminal(f *os.File) bool {
+	fi, err := f.Stat()
+	if err != nil {
+		return false
+	}
+
+	return (fi.Mode() & os.ModeCharDevice) != 0
+}
+
+func stdoutIsTerminal() bool { return isTerminal(os.Stdout) }
+func stderrIsTerminal() bool { return isTerminal(os.Stderr) }
+
+func shouldColorize() bool {
+	term := os.Getenv("TERM")
+	if term == "" || term == "dumb" {
+		return false
+	}
+
+	return stdoutIsTerminal() && stderrIsTerminal()
```

Comment:
Maybe it's better we just check stderr. WDYT @p1-0tr ?

Reply:
We need to check where `log.Writer()` goes, because - https://github.com/docker/model-runner/blob/main/pkg/inference/backends/llamacpp/llamacpp.go#L159.

Also we'd need to decorate `tailBuf`, here - https://github.com/docker/model-runner/blob/main/pkg/inference/backends/llamacpp/llamacpp.go#L160, to strip console control characters. Otherwise we'll end up sticking them in errors we return.